packages/schemas/src/setup/setup.ts

```
@@ -23,13 +23,132 @@ export const $ProductionReleaseInfo = z.object({
+});
+
+/**
+ * Resource-link hrefs render into `<a href>` on the login page, so they must be
+ * restricted to http(s) here (the server-side gate) — otherwise a crafted value
+ * like `javascript:` or `data:` would be a script-injection vector. The pattern
+ * also requires a dotted host (e.g. example.com) and mirrors the client form's.
+ */
+const RESOURCE_LINK_URL_PATTERN = /^https?:\/\/\S+\.\S+$/;
+
+/** A single resource link displayed in the branding panel. */
+export type ResourceLink = z.infer<typeof $ResourceLink>;
```

ESLint

[error] Export statements should appear at the end of the file

(import/exports-last)